src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -160,7 +160,10 @@ def to_matrix_with_info(x, dtype, copy=False):
       dims = len(x)
 
     d = np.zeros([dims])
-    out = np.array(x, dtype=dtype, copy=copy) # Try to avoid copy...
+    if copy:
+      out = np.array(x, dtype=dtype, copy=True)
+    else:
+      out = np.array(x, dtype=dtype, copy=None)
```

Comment:
Think the `None` option is [new in NumPy 2]( https://numpy.org/doc/stable/release/2.0.0-notes.html#new-copy-keyword-meaning-for-array-and-asarray-constructors ). It [isn't mentioned in NumPy 1]( https://numpy.org/doc/1.26/reference/generated/numpy.array.html ). So likely there is another condition needed to check for NumPy 2 and use the `copy` option then and if not drop the `copy` argument. That is assuming that is the behavior intended here

Reply:
Thanks!  I thought I could get away with it (and was depending on CI to tell me if I was wrong, which it did), but you're right.  Should be fixed in 57d2e8.